base/applications/rapps/loaddlg.cpp

```diff
@@ -557,6 +559,11 @@ CDownloadManager::DownloadDlgProc(HWND Dlg, UINT uMsg, WPARAM wParam, LPARAM lPa
             }
             return TRUE;
 
+        case WM_DESTROY:
+            g_ActiveOperationWindow = NULL;
+            g_Busy--;
```

Comment:
I wonder why you keep a counter `g_Busy` separated from `g_ActiveOperationWindow`.
Isn't it possible to only use the `g_ActiveOperationWindow` for doing the check in the `WM_CLOSE` handling in `CMainWindow::ProcessWindowMessage` (much above) ?

Reply:
No, you can have more than one download dialog open (RApps is somewhat buggy when you do this but it can happen).

Reply:
ok yeah i think this should be fixed later: if one DL dialog is already opened, and you add new other downloads to do, they get queued and the single existing DL dialog list gets refreshed.

Reply:
Yes I agree but there I do have to deal with threading, will look into it later.